File: yanwte-core/src/test/java/apiAcceptanceTests/dataExtensionPoint/DataExtensionInitializer.java
package apiAcceptanceTests.dataExtensionPoint;

import com.github.winteryoung.yanwte.DataExtensionPoint;
import com.github.winteryoung.yanwte.YanwteDataExtensionInitializer;
import org.jetbrains.annotations.NotNull;

/**
 * @author Winter Young
 * @since 2016/1/25
 */
public class DataExtensionInitializer extends YanwteDataExtensionInitializer {
    @NotNull
    @Override
    public Object initialize(@NotNull DataExtensionPoint dataExtensionPoint) {
        if (dataExtensionPoint instanceof TestData) {
            TestData testData = (TestData) dataExtensionPoint;
            return new TestDataExt(testData.getI());
        }
        throw new RuntimeException();
    }
}

File: yanwte-core/src/test/java/integrationTests/mixedTest/extspace1/DataExtensionInitializer.kt
package integrationTests.mixedTest.extspace1

import com.github.winteryoung.yanwte.DataExtensionPoint
import com.github.winteryoung.yanwte.YanwteDataExtensionInitializer
import integrationTests.mixedTest.TestData
import integrationTests.mixedTest.TestDataExt

class DataExtensionInitializer : YanwteDataExtensionInitializer() {
    override fun initialize(dataExtensionPoint: DataExtensionPoint): Any? {
        return when (dataExtensionPoint) {
            is TestData -> {
                TestDataExt(dataExtensionPoint.i)
            }
            else -> {
                null
            }
        }
    }
}